components/molecules/StackedAvatar/stacked-avatar.tsx

```diff
@@ -1,21 +1,36 @@
-import { getAvatarLink } from "lib/utils/github";
-import Avatar from "../../atoms/Avatar/avatar";
+import { useRouter } from "next/router";
 
+import AvatarHoverCard from "components/atoms/Avatar/avatar-hover-card";
+
+import useInsight from "lib/hooks/useInsight";
 export declare interface StackedAvatarProps {
   contributors: DbContribution[];
   visibleQuantity?: number;
 }
-const StackedAvatar = ({ contributors , visibleQuantity = 5}: StackedAvatarProps) => (
-  <div className="-space-x-3 flex">
-    {contributors && contributors.slice(0, visibleQuantity).map(({ host_login: hostLogin, name }, index) => (
-      <div
-        key={`contributor-avatar-${hostLogin}`}
-        className={`w-8 h-8 overflow-hidden rounded-full border-2 border-solid border-white z-${50-(index+1)*10}`}
-      >
-        <Avatar key={`${hostLogin + index}`} avatarURL={getAvatarLink(hostLogin)} alt={name} size={32} hasBorder isCircle />
-      </div>
-    ))}
-  </div>
-);
+
+const StackedAvatar = ({ contributors , visibleQuantity = 5}: StackedAvatarProps) => {
+  const router = useRouter();
+  const { filterName } = router.query;
+  const insightId = filterName as string;
+  const { data: insight, isError } = useInsight(insightId);
+  const repositories = insight ? insight.repos.map(repo => repo.repo_id) : [];
+
+  if (isError) {
+    console.log("Error fetching insight", isError);
+    return null;
```

Comment:
nit: This should return a valid element/error

```suggestion
    return <div>-</div>;
```